### PR TITLE
[WIP] Cirrus: Add "su" system-test matrix

### PR DESCRIPTION
**Pull request by cevich**
#### Does this PR introduce a user-facing change?

```release-note
Include "su" system testing in upstream CI
```

### CODE REVIEW

**Review by edsantiago — CHANGES_REQUESTED** [review]

**Comment by edsantiago** [comment]
Can you remind me why we need this? There's nothing in the commit message or PR. I vaguely remember you commenting on it in scrum last week, and saying that it's something I asked for, but I really don't think this is worth the complexity. We test `su` in gating tests, and yeah, that catches occasional `XDG` problems, but probably not enough to add so much more hassle to our CI.

File: contrib/cirrus/runner.sh [hunk 1]
```diff
@@ -481,18 +484,26 @@ if [[ "$PRIV_NAME" == "rootless" ]] && [[ "$UID" -eq 0 ]]; then
...
+        msg "************************************************************"
+        set -x
+        exec ssh $ROOTLESS_USER@localhost \
+                -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no \
+                -o CheckHostIP=no $GOSRC/$SCRIPT_BASE/runner.sh
+    elif [[ "$PRIV_NAME" == "su" ]]; then
+        msg "Re-executing runner through su as user '$ROOTLESS_USER'"
+        msg "************************************************************"
+        set -x
+        exec su $ROOTLESS_USER $GOSRC/$SCRIPT_BASE/runner.sh
```
[the remark this answers is not in the document]
Comment (cevich) [reply contrib/cirrus/runner.sh:503]: With `-c` and `-` there's probably no point in running this test vs the `ssh` variant.

File: contrib/cirrus/runner.sh [hunk 2]
```diff
@@ -481,18 +484,26 @@ if [[ "$PRIV_NAME" == "rootless" ]] && [[ "$UID" -eq 0 ]]; then
...
-    msg "Re-executing runner through ssh as user '$ROOTLESS_USER'"
-    msg "************************************************************"
-    set -x
-    exec ssh $ROOTLESS_USER@localhost \
-            -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no \
-            -o CheckHostIP=no $GOSRC/$SCRIPT_BASE/runner.sh
-    # Does not return!
+    # N/B: Nothing below will return!
+    if [[ "$PRIV_NAME" == "rootless" ]]; then
+        msg "Re-executing runner through ssh as user '$ROOTLESS_USER'"
```
[the remark this answers is not in the document]
Comment (cevich) [reply contrib/cirrus/runner.sh:493]: Good suggestion.

**Comment by cevich** [comment]
Ugh:

`+ exec su some12599dude /var/tmp/go/src/github.com/containers/podman/./contrib/cirrus/runner.sh
/usr/share/automation/lib/console_output.sh: line 70: /dev/stderr: Permission denied`

That's what I was afraid might happen.

**Comment by cevich** [comment]
> Can you remind me why we need this? There's nothing in the commit message or PR.

Because the team asked for it.  Apparently `su podman foo` is something that's suppose to work and therefore should be tested.  Perhaps it's something that works in docker?  In any case, with podman and buildah, the tty & pty ownership problem is a long-standing problem (i.e. using su or sudo inside a Dockerfile also breaks in this way).

> We test su in gating tests, and yeah, that catches occasional XDG problems, but probably not enough to add so much more hassle to our CI.

That was part of the other reason IIRC, the team wanted to catch these problems earlier in the development cycle.

Anywho...I agree, the extra complexity and gymnastics is probably not worth it in upstream CI.  I'll bring this up in SCRUM.

**Comment by edsantiago** [comment]
I'm super confused. Do you mean `sudo`? `su user foo` does not work, has never worked in all the years I've been around. You either do `sudo user foo`, or `su user -c "foo"`.

Anyhow, apart from that, yeah, let's figure out what problem we're trying to solve. It might be pretty easy to just add a test that runs `su/sudo` and tests a handful of likely-to-fail podman commands.

**Comment by cevich** [comment]
> su user foo does not work

Woops, yeah, I meant sudo.  In any case, `-l` or `-` is still going to result in `/dev/stderr: permission denied` all over the place.

> It might be pretty easy to just add a test that runs su/sudo and tests a handful of likely-to-fail podman commands.

Meh, I'd say probably even that isn't worth the extra complexity needed to support it.

**Comment by Luap99** [comment]
see https://github.com/containers/podman/issues/14603

su or sudo are not supposed to work since they do not create a proper login session and without the login option even have the wrong XDG vars set.

As @edsantiago pointed out (https://github.com/containers/podman/pull/14992#discussion_r925812030) you need to enable lingering and the use `su -l ...` or `su - ...` to make it work.

**Comment by cevich** [comment]
> you need to enable lingering and the use su -l ... or su - ... to make it work.

Thanks Paul.  Yeah, and that totally beggs the issue, since at that point we're not very different from the `rootless` (ssh-based) environment.  I'm with Ed, I think we should abandon this effort.  A singular test for the #14603 fix should be more than sufficient.

**Comment by rhatdan** [comment]
I am fine with that.  Although I would want to hear @vrothberg @baude @mheon @giuseppe opinion.

**Comment by mheon** [comment]
My understanding is that we began this because the RHEL gating tests use `su` and we wanted to try and catch problems in them in CI. However, Paul is entirely correct that we do not expect sudo/su to work as a general principle. Given that, I'm OK with aborting this effort; dealing with problems as they arise in RHEL CI is inconvenient, but testing something that should never be used in production on every pull request seems worse.

**Comment by giuseppe** [comment]
I agree with Matt.  Let's not test stuff that is not even supposed to work.

**Comment by cevich** [comment]
SGTM, thanks everyone for considering this.

**Pull request closed without merging**